config/config_operation_wrapper.go

```diff
@@ -0,0 +1,46 @@
+package config
+
+import (
+	op "imagecore/operation" // Grab `EncoderOption` from operation package.
+)
+
+// This ts a utility function to convert pipeline block to image operation.
+
+func PipelineBlockToOperation(pb PipelineBlock) op.Operation {
+
+	// We assume the loaded config has been fully checked.
+	// Therefore we don't return error here.
+	switch pb.Operation {
+
+	case "decode":
+		return op.Decode()
+
+		// - `icc_embed`
+
+		// - `write`
+	case "resize":
+		if pb.Resize.Factor != 0.0 {
```

Comment:
`Switch` is a cool way to deal with multiple conditions in GoLang. It can make code cleaner.
But with some usage such as nested-if, we might still stick with if-else clause. : D